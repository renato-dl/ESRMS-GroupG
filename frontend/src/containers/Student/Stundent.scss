.student {
    &_grades{
        font-size: 20px;
        background-color:yellow;
        position:relative;
        top:10px;
        left:30px;
    }
}
