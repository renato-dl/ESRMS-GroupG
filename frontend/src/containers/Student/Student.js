import React from 'react';
import { StudentCard } from '../../components/StudentCard/StudentCard';
import './Stundent.scss';
import {ApplicationStoreContext} from '../../store';

export class Student extends React.Component {
  static contextType = ApplicationStoreContext;

  showMarks = async () => {
    this.props.history.push(`/parent/student/${this.props.match.params.studentID}/marks`);
  };

  render() {
    const {selectedStudent} = this.context.state.parent;

    return (
      <div className="student-container">
        <h2 className="title">Students place in here</h2>
        <StudentCard {...selectedStudent} />

        <button className='student_grades' type='button' onClick={this.showMarks} >
          Show grades
        </button>

      </div>
    )
  }
}