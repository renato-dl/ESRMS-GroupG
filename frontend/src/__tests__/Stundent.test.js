import React from 'react';
import {Student} from '../containers/Student/Student';
import renderer from 'react-test-renderer';

describe('Testing Student component', () => {
  
  test('Testing snapshot', () => {
    const component = renderer.create(
      <Student />
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

})
