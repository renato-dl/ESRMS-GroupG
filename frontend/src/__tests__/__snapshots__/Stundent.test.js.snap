// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Student component Testing snapshot 1`] = `
<div
  className="student-container"
>
  <h2
    className="title"
  >
    Students place in here
  </h2>
  <div
    className="ui card student"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        src="student-stub.png"
      />
    </div>
    <div
      className="content"
    >
      <div
        className="header"
      >
         
      </div>
    </div>
    <div
      className="extra content"
    >
      <div>
        SSN: 
      </div>
      <div>
        BirthDate: 
        Invalid Date
      </div>
    </div>
  </div>
  <button
    className="student_grades"
    onClick={[Function]}
    type="button"
  >
    show grades
  </button>
</div>
`;
